### PR TITLE
DropdownButton title can be any React node

### PR DESCRIPTION
**Pull request by karolklp**
- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
react-bootstrap/react-bootstrap#928

### CODE REVIEW

File: react-bootstrap/index.d.ts [hunk 1]
```diff
@@ -590,6 +590,7 @@ declare namespace ReactBootstrap {
         active?: boolean;
         noCaret?: boolean;
         eventKey?: any;
+        title?: any; // TODO: Add more specific type
```
Comment (vsiao) [inline react-bootstrap/index.d.ts]: `React.ReactNode`? I appreciate the `TODO`, but in practice that will never get fixed.

Comment (karolklp) [inline react-bootstrap/index.d.ts]: Yep, I've tried it but I have no idea why it doesn't accept the code that works ;)
`<NavDropdown eventKey={3} title={<i className="glyphicon glyphicon-user">Dropdown</i>} id='basic-nav-dropdown'>`
and I recieve compilation error `"Type Element is not assignable to Type 'Element' is not assignable to type 'string | (number & string) | (true & string) | (false & string) | ({} & string) | (ReactElement<a...'.`

File: react-bootstrap/index.d.ts [hunk 2]
```diff
@@ -590,6 +590,7 @@ declare namespace ReactBootstrap {
...
     }
     type NavDropdownProps = NavDropdownBaseProps & React.HTMLProps<NavDropdown>;
```
Comment (vsiao) [inline react-bootstrap/index.d.ts]: This is your problem here: by intersecting with `HTMLProps` you get `title?: ReactNode & string`, to which no value is assignable (except maybe `null`)

Comment (bowdenk7) [inline react-bootstrap/index.d.ts]: @karolklp can you address the feedback here so we can get this merged? Thanks!

Comment (karolklp) [inline react-bootstrap/index.d.ts]: @vsiao: So it should be union? Something like `title?: ReactNode|string`? Or does it need separate type like this?

Comment (minestarks) [inline react-bootstrap/index.d.ts]: I'm totally stumped by this, my unfamiliarity with react isn't helping. @DanielRosenwasser would you know what to do here?

**Comment by paulvanbrenk** [comment]
It doesn't look like we're making any progress here. closing for now, please re-open when there is a good solution.

**Comment by darewreck54** [comment]
Can't you just make it title?:any until a better solution is found.  This at least gives you the ability to use the an image in the title for typescript application for the navdropdown?

**Pull request closed without merging**